src/tanxconnector/utils.py

```diff
@@ -118,8 +119,31 @@ def filter_cross_chain_coin(config, coin, type):
     else:
         raise CoinNotFoundError('Type not found')
     if not allowed_token:
-        raise CoinNotFoundError(f'Coin {coin} not found')
+        raise CoinNotFoundError(f'Coin {coin} not found, allowed tokens: {list(allowed_tokens.keys())}')
 
     current_coin = allowed_tokens[coin]
     return current_coin
 
+def get_native_currency_by_network(network: str) -> str:
+    """
+    Get the native currency of a given network.
+
+    Args:
+    network (str): The name of the network.
+
+    Returns:
+    str: The native currency of the specified network.
+
+    Raises:
+    KeyError: If the provided network is not in the supported networks.
+    """
+
+    network_to_native_currency = {
```

Comment:
The problem with hardcoding `network_to_native_currency` is we have to release new version of package each time a network is added in our system. There is a chance that this gets missed which will lead to breaking of code.

What we can do is dynamically get this data from an endpoint so that it is future proof. Also till the time that endpoint is added, we should have a default value so that code doesn't break. 

Default can be `eth` as that is most common native token